src/test/java/seedu/internhunter/ui/CardTest.java

```diff
@@ -0,0 +1,262 @@
+            assertTrue(applicationCardWithStatusApplied.equals(applicationCardWithStatusApplied));
+            assertTrue(applicationCardWithStatusAccepted.equals(applicationCardWithStatusAccepted));
+            assertTrue(applicationCardWithStatusInterview.equals(applicationCardWithStatusInterview));
+            assertTrue(applicationCardWithStatusWaiting.equals(applicationCardWithStatusWaiting));
+            assertTrue(applicationCardWithStatusRejected.equals(applicationCardWithStatusRejected));
+            assertTrue(applicationCardWithStatusOffered.equals(applicationCardWithStatusOffered));
+
+            // null -> return false
+            assertFalse(applicationCardWithStatusApplied.equals(null));
+            assertFalse(applicationCardWithStatusAccepted.equals(null));
+            assertFalse(applicationCardWithStatusInterview.equals(null));
+            assertFalse(applicationCardWithStatusWaiting.equals(null));
+            assertFalse(applicationCardWithStatusRejected.equals(null));
+            assertFalse(applicationCardWithStatusOffered.equals(null));
+
+            // different type -> return false
+            assertFalse(applicationCardWithStatusApplied.equals(0.5f));
+            assertFalse(applicationCardWithStatusAccepted.equals(0.5f));
+            assertFalse(applicationCardWithStatusInterview.equals(0.5f));
+            assertFalse(applicationCardWithStatusWaiting.equals(0.5f));
+            assertFalse(applicationCardWithStatusRejected.equals(0.5f));
+            assertFalse(applicationCardWithStatusOffered.equals(0.5f));
+
+            /* Test for company card. The 3 methods that are being tested are getItem, getDisplayedIndex
+             * and the equals method */
+
+            // test for getItem for company card
+            assertEquals(companyItemBuilder.build(), companyCard.getItem());
+            assertEquals(companyItemBuilderWithIndustries.build(), companyCardWithIndustries.getItem());
+
+            // test for getDisplayedIndex for company card
+            assertEquals(1, companyCard.getDisplayedIndex());
+            assertEquals(2, companyCardWithIndustries.getDisplayedIndex());
+
+            // test equals for company card
+
+            // same value -> return true
+            assertTrue(companyCard.equals(new CompanyCard(companyItemBuilder.build(), 1)));
+            assertTrue(companyCardWithIndustries.equals(new CompanyCard(companyItemBuilderWithIndustries.build(), 2)));
+
+            // same object -> return true
+            assertTrue(companyCard.equals(companyCard));
+            assertTrue(companyCardWithIndustries.equals(companyCardWithIndustries));
+
+            // null -> return false
+            assertFalse(companyCard.equals(null));
+            assertFalse(companyCardWithIndustries.equals(null));
+
+            // different type -> return false
+            assertFalse(companyCard.equals(0.5f));
+            assertFalse(companyCard.equals(0.5f));
```

Comment:
Should this be using `companyCardWithIndustries`?

Reply:
good catch resolved